Training4/src/task-1/src/App.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {useEffect, useReducer, useState} from 'react';
+import './App.css';
+import {Joke} from "./interfaces/joke";
+import axios, {AxiosResponse} from "axios";
+import JokeCard from "./component/joke-card";
+import {jokeReducer} from "./reducer/joke-reducer";
+import throttle from "lodash.throttle";
+const base_url = 'https://official-joke-api.appspot.com/random_ten'
+
+type StateType = {
+    data?: Joke[],
+    isLoading?: boolean,
+    error?: any
+}
+
+const initState: StateType = {
+    data: [],
+    error: null
+}
+
+function App() {
+    const [isLoading, setLoading] = useState<boolean>(true)
+    const [jokes, dispatch] = useReducer(jokeReducer, initState)
+    useEffect(() => {
+        fetchJokes()
+    }, [])
+
+    const fetchJokes = () => {
+        setLoading(true);
+        axios.get(base_url)
+            .then((res: AxiosResponse<Joke[]>) => {
+                setLoading(false);
+                dispatch({type: 'SUCCESS', data: [res.data[0], res.data[2]]})
+            })
+            .catch((err) => {
+                dispatch({type: 'ERROR', data: err})
+            })
+    }
+
+    const handleGetMore = throttle(() => {
```

Comment:
still wrong, arrow function will be created new, handleGetMore function also created new

Reply:
we need to memorize the fetchJokes first, should not let create every time we got render -> hint: use callback dependencies [setLoading, dispatch]  -> it will be same [], because setLoading & dispatch are not changed at anytime

then memorize the handleGetMore